### DISCUSSION TITLE
Seeking HTML and React Autocompletion Solutions

### DISCUSSION THREAD

**Discussion by n4vn337** [LazyVim/LazyVim#2505] [Q&A] [answered]
I've just set up lazyvim in my nvim, and overall, it's functioning well. However, I'm encountering an issue where HTML snippet autocompletion and suggestions for React projects aren't working as expected. It's worth noting that these features work perfectly fine in .html files. Any guidance on what I might be overlooking or need to add to ensure proper functionality?

As an alternative, I also experimented with 'mattn/emmet-vim,' and it performs effectively. However, I find it a bit cumbersome as I have to use <C-Y> each time I want it to work. This slows down the process, especially for common tags like 'button'. Is there a way to streamline this and avoid the need for manual keystrokes?

**Comment by n4vn337** [comment]
#1142 is on similar topic I believe, I tried creating a new file inside plugin and return that code snippet, didnt work for me. Am I missing something here?

**Reply by dpetka2001** [reply, marked as answer]
Try this
```lua
-- ~/.config/nvim/lua/plugins/snippets.lua
return {
  {
    "rafamadriz/friendly-snippets",
    config = function()
      require("luasnip").filetype_extend("javascriptreact", { "html" })
      require("luasnip.loaders.from_vscode").lazy_load()
    end,
  },
}
```
For some reason it gives duplicate entries from snippets in the `nvim-cmp` auto-completion menu, but that doesn't have anything to do with LazyVim. It's some kind of regression either with Luasnip or nvim-cmp itself. So, if it annoys you, you should create an issue in one of those repos.

**Reply by n4vn337** [reply]
Thanks this works, for me. Do you have any Idea how can i configure rafc snippet, currently it allows rafc snippet only?

**Reply by dpetka2001** [reply]
I don't know what rafc snippet is. Sorry.
